pkg/services/accesscontrol/plugins.go

```diff
@@ -0,0 +1,31 @@
+package accesscontrol
+
+import (
+	"strings"
+)
+
+// ValidatePluginPermissions errors when a permission does not match expected pattern for plugins
+func ValidatePluginPermissions(pluginID string, permissions []Permission) error {
+	for i := range permissions {
+		if !strings.HasPrefix(permissions[i].Action, AppPluginRolePrefix) &&
+			!strings.HasPrefix(permissions[i].Action, pluginID+":") &&
+			!strings.HasPrefix(permissions[i].Action, pluginID+".") {
+			return &ErrorActionPrefixMissing{Action: permissions[i].Action, Prefixes: []string{AppPluginRolePrefix, pluginID + ":"}}
```

Comment:
nit: should this be:
```suggestion
			return &ErrorActionPrefixMissing{Action: permissions[i].Action, Prefixes: []string{AppPluginRolePrefix, pluginID + ":", pluginID+"."}}
```